drivers/sensor/adxl345/adxl345.c

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright (c) 2020 Antmicro <www.antmicro.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/sensor.h>
+#include <init.h>
+#include <drivers/gpio.h>
+#include <drivers/i2c.h>
+#include <logging/log.h>
+
+#include "adxl345.h"
+
+LOG_MODULE_REGISTER(ADXL345, CONFIG_SENSOR_LOG_LEVEL);
+
+static int adxl345_read_reg(struct device *dev, u8_t reg)
+{
+	struct adxl345_dev_data *data = dev->driver_data;
+	u8_t reg_val;
+	int rc = i2c_write_read(data->i2c_master,
+				data->i2c_addr,
+				&reg, 1,
+				&reg_val, 1);
+
+	if (rc < 0) {
+		LOG_ERR("Register read failed with rc=%d\n", rc);
+		return rc;
+	}
+
+	return reg_val;
+}
+
+static int adxl345_write_reg(struct device *dev, u8_t reg, u8_t val)
```

Comment:
Please use `i2c_reg_write_byte` and `i2c_reg_read_byte` directly. This function and `adxl345_read_reg` are unnecessary wrappers.

Reply:
Done